code/modules/mining/ores_coins.dm

```diff
@@ -324,6 +324,8 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 	var/cooldown = 0
 	var/value = 1
 	var/coinflip
+	var/is_stack = 0
```

Comment:
You're going to hate me for this, but

rather than making a second stack type class, with its own unique handling, just make coins a subclass of stacks
you will need to update things (like vendors) that currently take coins, and have them take 1 from the stack amount

then in that subclass, add the flipping, etc if there is only 1 coin

painful now, but it saves us from having two stack classes to maintain

Reply:
Heck.

Reply:
I kinda didn't want to use base stacks. They only have 3 variants (I want to use 6) and have a bunch of material construction related baggage, that's why I avoided them. I would also have to change the coin press machine to make stacks for some coin types and single coins for others. Idk about this...

Reply:
![image](https://user-images.githubusercontent.com/37093159/48326013-258f7000-e5ec-11e8-9dd5-d75a947d528e.png)

Reply:
![image](https://user-images.githubusercontent.com/37093159/48326033-3a6c0380-e5ec-11e8-9be2-f579feb99292.png)

Reply:
Ill have a squizz when home and see what we can do

worst case is we can make a stack/currency/ child class, with a return override for the various crafting things.

just dont want to double up on the various duplication, merging, etc. handling

Reply:
Makes sense. Good luck!